x-pack/plugins/maps/common/constants.ts

```diff
@@ -73,6 +73,22 @@ export enum FIELD_ORIGIN {
   SOURCE = 'source',
   JOIN = 'join',
 }
+export const JOIN_FIELD_NAME_PREFIX = '__kbnjoin__';
+
+// function in common since its needed by migration
+export function getJoinAggKey({
+  aggType,
+  aggFieldName,
+  rightSourceId,
+}: {
+  aggType: AGG_TYPE;
+  aggFieldName?: string;
+  rightSourceId: string;
```

Comment:
Can not pass in ISource. Source class instances live in public. This method lives in common because its used in migrations which live in server and public.